test/xkeyboard-config-test.py.in

```diff
@@ -1,97 +1,187 @@
 #!/usr/bin/env python3
+
+from __future__ import annotations
+
 import argparse
+import dataclasses
+import itertools
 import multiprocessing
-import sys
-import subprocess
 import os
+import subprocess
+import sys
 import xml.etree.ElementTree as ET
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
 from pathlib import Path
-
-
-verbose = False
+from typing import (
+    Any,
+    ClassVar,
+    Generator,
+    Iterable,
+    NoReturn,
+    Protocol,
+    TextIO,
+    TypeVar,
+    cast,
+)
+
+try:
+    # Python >=3.11 only
+    from typing import Self
+except ImportError:
+    Self = Any
+
+WILDCARD = "*"
 
 DEFAULT_RULES_XML = "@XKB_CONFIG_ROOT@/rules/evdev.xml"
 
 # Meson needs to fill this in so we can call the tool in the buildir.
 EXTRA_PATH = "@MESON_BUILD_ROOT@"
-os.environ["PATH"] = ":".join([EXTRA_PATH, os.getenv("PATH")])
-
+os.environ["PATH"] = ":".join(filter(None, (EXTRA_PATH, os.getenv("PATH"))))
 
-def escape(s):
-    return s.replace('"', '\\"')
 
+@dataclass
+class RMLVO:
+    rules: str
+    model: str
+    layout: str
+    variant: str | None
+    option: str | None
 
-# The function generating the progress bar (if any).
-def create_progress_bar(verbose):
-    def noop_progress_bar(x, total, file=None):
-        return x
-
-    progress_bar = noop_progress_bar
-    if not verbose and os.isatty(sys.stdout.fileno()):
-        try:
-            from tqdm import tqdm
-
-            progress_bar = tqdm
-        except ImportError:
-            pass
+    @property
+    def __iter__(self) -> Generator[str | None, None, None]:
```

Comment:
```suggestion
    def __iter__(self) -> Iterator[str | None]:
```

(the fact that it's a generator is an implementation detail)

Reply:
Fixed. Ref doc: https://docs.python.org/3/library/typing.html#annotating-generators-and-coroutines